File: simplerpc/src/main/java/com/springmarker/simplerpc/pojo/ExchangeRequest.java
package com.springmarker.simplerpc.pojo;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * 定义一个基本通讯Request对象。
 *
 * @author Springmarker
 * @date 2018/10/24 23:02
 */
@Data
@AllArgsConstructor
@NoArgsConstructor
public class ExchangeRequest {

    private int clientId;
    private int id;
    private RpcRequest rpcRequest;


}

File: simplerpc/src/main/java/com/springmarker/simplerpc/pojo/ExchangeRequest.kt
package com.springmarker.simplerpc.pojo

/**
 * 定义一个基本通讯Request对象。
 *
 * @author Frank
 * @date 2018/10/24 23:02
 */
class ExchangeRequest(
        val clientId: Int,
        val id: Int,
        val rpcRequest: RpcRequest
) {

    class RpcRequest(
            val methodHashCode: Int,
            val paramList: List<Any>,
            val needReturn: Int
    )

}